scratch/dumbbell.cc

```diff
@@ -0,0 +1,441 @@
+                                 "Be mindful that both left and right edge will have the same delay. " 
+                                 "Edges will have the default delay if not specified. " 
+                                 "(e.g. \"1000,500,1000,2000\")";
+
+  CommandLine cmd;
+  cmd.AddValue ("bandwidth_Mbps", "Bandwidth for the bottleneck router (Mbps).", bwMbps);
+  cmd.AddValue ("delay_us", "Bottleneck link delay (us).", btlDelUs);
+  cmd.AddValue ("queue_capacity_p", "Router queue size at bottleneck router (packets).", btlQueP);
+  cmd.AddValue ("edge_delay_us", edge_delay_usage, edgeDelayUs);
+  cmd.AddValue ("packet_size", "Size of a single packet (bytes)", packet_size);
+  cmd.AddValue ("experiment_duration_s", "Simulation duration (s).", durS);
+  cmd.AddValue ("warmup_s", "Time before delaying ACKs (s)", warmupS);
+  cmd.AddValue ("pcap", "Record a pcap trace from each port (true or false).", pcap);
+  cmd.AddValue ("csv", "Record a csv file for BBR receiver (true or false).", csv);
+  cmd.AddValue ("model", "Path to the model file.", modelFlp);
+  cmd.AddValue ("out_dir", "Directory in which to store output files.", outDir);
+  cmd.AddValue ("unfair_flows", "Number of BBR flows.", unfairFlows);
+  cmd.AddValue ("other_flows", "Number of non-BBR flows.", otherFlows);
+  cmd.AddValue ("other_proto", "The TCP variant to use (e.g., \"ns3::TcpCubic\") for the non-BBR flows.", otherProto);
+  cmd.AddValue ("enable", "Enable unfairness mitigation (true or false).", enableUnfair);
+  cmd.AddValue ("fair_share_type", "How to estimate the bandwidth fair share.", fairShareType);
+  cmd.AddValue ("ack_pacing_type", "How to estimate ACK pacing interval.", ackPacingType);
+  cmd.Parse (argc, argv);
+
+  // Verify the protofol specified for the non-BBR flows.
+  NS_ABORT_UNLESS (otherProto == "ns3::TcpNewReno" ||
+                   otherProto == "ns3::TcpCubic" ||
+                   otherProto == "ns3::TcpBbr");
+
+  uint32_t rttUs = (btlDelUs + EDGE_DELAY_US * 2) * 2;
+
+  // Number of Nodes
+  uint32_t num_nodes = unfairFlows + otherFlows;
+
+  // Bandwidth
+  std::stringstream bwSs;
+  bwSs << bwMbps << "Mbps";
+  std::string bw = bwSs.str ();
+
+  // Delay
+  std::stringstream btlDelSs;
+  btlDelSs << btlDelUs << "us";
+  std::string btlDel = btlDelSs.str ();
+
+  // Queue size
+  std::stringstream btlQueSs;
+  btlQueSs << btlQueP << "p";
+  std::string btlQue = btlQueSs.str ();
+
+  double routerToDstBW = bwMbps;
+  std::stringstream sndSS;
+  sndSS << routerToDstBW << "Mbps";
+  std::string btlBWStr = sndSS.str ();
+
+  mtu = packet_size + HEADER_AND_OPTION;
+
+  // Edge delays
+  std::vector<uint32_t> edge_delays;
+  std::stringstream ss(edgeDelayUs);
+
+  while (ss.good()) {
+    std::string edge_delay;
+    getline(ss, edge_delay, ',');
+    if (!edge_delay.empty()) {
+      edge_delays.push_back(std::stoi(edge_delay));
+    }
+  }
```

Comment:
I was thinking that the number of delays could be less than the number of nodes. (e.g. if there are for nodes and the user specifies 100,200, the last two nodes will have the default delay of 500ms). 

I will make it to enforce that input has the same length of the number of nodes. For shortcut mode, it will accept '[delay]' for all node pairs.